addons/stock/product.py

```diff
@@ -402,6 +415,18 @@ def action_view_routes(self, cr, uid, ids, context=None):
         result['domain'] = "[('id','in',[" + ','.join(map(str, route_ids)) + "])]"
         return result
 
+    def onchange_track_all(self, cr, uid, ids, track_all, context=None):
+        warning = {}
+        if ids and track_all:
+            quant_ids = self.pool.get('stock.quant').search(cr, uid, [('product_id.product_tmpl_id','in',ids), ('lot_id','=', False), ('location_id.usage','=', 'internal')])
+            if quant_ids:
+                warning_msgs =  _("There are Quants without lot for this product. You need to serialize them first to prevent inventory loss.")
```

Comment:
I don't think you would use 'serialize' in this case.  I would prefer 'assign a serial number (e.g. by creating an inventory)'